### PR TITLE
#20 Admin can create and delete moderators

### PR DESCRIPTION
**Pull request by mrkava**
- [ ] @Kmalynss 
- [ ] @f1nwe 
- [ ] @igorkasyanchuk

- Admin can create moderator
- Admin can delete moderator
- Basic view for create moderator form
- Rspecs for all actions

### CODE REVIEW

File: app/views/account/users/_form.slim [hunk 1]
```diff
@@ -0,0 +1,11 @@
+div
+  = form_for(@user, url: yield(:form_path)) do |f|
```
Comment (igorkasyanchuk) [inline app/views/account/users/_form.slim]: simple_form

File: app/views/account/users/new.slim [hunk 2]
```diff
@@ -0,0 +1,10 @@
+- provide(:form_path, account_users_path)
+h1 Create new moderator
```
Comment (igorkasyanchuk) [inline app/views/account/users/new.slim]: use render partial, path: path

REMOVE YIELD from form

File: app/views/account/users/new.slim [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+= render 'form'
+
+- if @user.errors.any?
+  div
+    h3= pluralize(@user.errors.count, "error")
```
Comment (igorkasyanchuk) [inline app/views/account/users/new.slim]: use dynamic_form gem

File: app/views/account/users/_form.slim (hunk 1 above)
Comment (mrkava) [inline app/views/account/users/_form.slim]: fixed

File: app/views/account/users/new.slim (hunk 2 above)
Comment (mrkava) [inline app/views/account/users/new.slim]: fixed

File: app/views/account/users/new.slim (hunk 3 above)
Comment (mrkava) [inline app/views/account/users/new.slim]: fixed

File: app/controllers/account/users_controller.rb [hunk 4]
```diff
@@ -1,19 +1,50 @@
 # frozen_string_literal: true
 module Account
   class UsersController < ApplicationController
+    before_action :authenticate_user!
+    before_action :set_user, only: [:destroy]
```
Comment (Kmalynss) [inline app/controllers/account/users_controller.rb]: I belive that this fillter is needed for ```edit``` and ```update``` actions as well.

File: app/views/account/users/index.slim [hunk 5]
```diff
@@ -1,4 +1,10 @@
 h1 All Users
-ul.users
+table.users
```
Comment (igorkasyanchuk) [inline app/views/account/users/index.slim]: use bootstrap table styles. It has many and remove this style

File: app/controllers/account/users_controller.rb (hunk 4 above)
Comment (skcc321) [inline app/controllers/account/users_controller.rb]: It is a bad practice to define instance variable in private method.
please remove this before action, rename set_user to resource, and add @user = resource in destroy method and everywhere where you need @user.

File: app/views/account/users/index.slim (hunk 5 above)
Comment (mrkava) [inline app/views/account/users/index.slim]: done

File: app/controllers/account/users_controller.rb (hunk 4 above)
Comment (mrkava) [inline app/controllers/account/users_controller.rb]: done

Comment (mrkava) [inline app/controllers/account/users_controller.rb]: refactored like @skcc321 said - https://github.com/Kotany-IF070-WebUI-RoR/NewVasjuky/pull/33#pullrequestreview-14388655

so we don't use this filter now

File: app/controllers/account/users_controller.rb [hunk 6]
```diff
@@ -1,19 +1,50 @@
...
+      end
+    end
 
     def edit; end
 
     def update; end
 
-    def destroy; end
+    def destroy
+      @user = resource
```
Comment (Kmalynss) [inline app/controllers/account/users_controller.rb]: It looks like ```authorize @user``` should be moved to before filter.

Comment (mrkava) [inline app/controllers/account/users_controller.rb]: But we have different @user for each action, which is not defined **before**...

**Pull request closed without merging**